core/trino-main/src/main/java/io/trino/operator/OperatorStats.java

```diff
@@ -781,4 +794,50 @@ public OperatorStats withPipelineMetrics(Metrics pipelineMetrics)
                 blockedReason,
                 info);
     }
+
+    public OperatorStats addConnectorSplitSourceMetrics(Metrics splitSourceMetrics)
```

Comment:
it's not add since you are returning a new instance: withConnectorSplitSourceMetrics is more appropriate